tools/src/main/java/it/eng/tools/client/rest/OkHttpRestClient.java

```diff
@@ -109,4 +117,37 @@ public GenericApiResponse<String> sendGETRequest(String targetAddress, String au
 			return GenericApiResponse.error(e.getLocalizedMessage());
 		}
 	}
+	
+	public String sendInternalRequest(String contextAddress, HttpMethod method, JsonNode jsonBody) {
```

Comment:
Idea here is to have 2 methods, one that sends protocol request, can be GET or POST, based on parameter and other, internal API call, that will hide internal api address, that will be out of reach for developer or connector user. (can be get from properties).
Another improvement is that we return expected object from method call, wrapped with GenericApiResponse<T> instead returning GenericApiResponse<String> and then handle it afterwards. Second solution is still POC/under investigation how to do it.